auth/guard.go

```diff
@@ -0,0 +1,47 @@
+package auth
+
+import (
+	"log"
+	"net/http"
+	"regexp"
+)
+
+// APIMiddleware authenticates API requests
+func APIMiddleware(rw http.ResponseWriter, request *http.Request, next http.HandlerFunc) {
+	authHeader := request.Header.Get("Authorization")
+	if authHeader == "" {
+		rw.WriteHeader(401)
+		rw.Write([]byte("not authorized"))
+		return
+	}
+
+	if !validateAuthHeader(authHeader) {
+		log.Println("Authorization header is invalid:", authHeader)
+		rw.WriteHeader(401)
+		rw.Write([]byte("invalid authorization header"))
+		return
+	}
+
+	if !validateJWTToken(authHeader) {
+		rw.WriteHeader(401)
+		rw.Write([]byte("not authorized"))
```

Comment:
Just a temporary response right now. API endpoints should return standardized JSON once we get there. Creating a task for it